activelearning/dataset/grid.py

```diff
@@ -48,7 +48,7 @@ def append(self, X, y):
         append new instances to the data
         """
         # X, y = self.deprocess(X, y) # append data in raw form (i.e., not normalized)
-        X *= self.grid_size
+        X = (X + 1) * self.grid_size / 2  # denormalize from the range [-1; 1]
```

Comment:
I suspect this will not work as expected in general (for example for both Branin and Hartmann). Can you elaborate what the intention is here?

Reply:
without normalization we would have grid values in the range of [0; grid_size] to scale between [0; 1] we would just divide by grid_size. to scale between [-1; 1] (as we need to in the new version) we can divide by half the grid size and subtract 1. 

if we want to add new selected datapoints (which are selected between [-1; 1]), and add it to the base data that is not normalized, we also have to denormalize the new data to be in the original range of [0; grid_size], which is the inverse of the above (add 1 and multiply by half the grid_size)

does that make sense? why would you assume that it does not work like this?

Reply:
Ok, this is going to need some additional documentation because even I am confused. And in the future, we will have data from other environments and it is very important to know in which format the input data is and in which format the outputs should be.

Reply:
Also, unclear what "raw" means. In the GFlowNet repo, we have these formats for states/data:
- "Environment format": the representation of the states in a GFlowNet environment. For example, for the grid, that is `[0, length]`.
- Proxy format: the representation received by a Proxy, or an Oracle in the active learning sense. That's the output of `Env.states2proxy(states)`.
- Policy format: the representation of the states passed to the policy models.
- Readable: a string representation meant to be more "human-readable"

Reply:
I would use these terms if possible.

And more importantly, I would (re-)use the methods from the GFlowNet environment to do the conversions.

Reply:
So, this said, let's try to clarify:
- In what format are the inputs?
- In what format should the outputs be?

Reply:
- the inputs are in the range of [0; grid_size] 
- the outputs in the range [-1; 1]

I am not sure if we should use the term "proxy" at all in this context... in this active learning repo we only have the components "surrogate", "acquisition", "sampler", "selector", and "oracle". I think we should keep the terms more generic and not specific to the GFN task

Reply:
or do you want to generally introduce proxies into the active learning workflow? currently it is only a thing when using GFNs as sampler

Reply:
Fine with me, but it's important to keep in mind that Oracle = GFN Proxy, if we use GFN. We can use the term "Oracle format" if you prefer. Then, we have the following:
- Inputs are in "environment format"
- Outputs are in "oracle format"
Do you agree? If so, I would document that. And like I said above I would reuse the method `states2proxy()`.

Reply:
> or do you want to generally introduce proxies into the active learning workflow? currently it is only a thing when using GFNs as sampler

Well, not only. When you use Branin (for example), you are using a GFN proxy, even if you don't use GFN as the sampler. See here: https://github.com/alexhernandezgarcia/activelearning/blob/ch/updateGFN/activelearning/oracle/oracle.py#L24

Reply:
hmm... right

I'll have to think about how to best structure it, but ideally with a clear head and nose^^